shared/interfaces.go

```diff
@@ -0,0 +1,13 @@
+package shared
```

Comment:
I agree that "shared" could become a dumping ground. However, I disagree that this is for deps.

Interfaces (and currently a TChannel util function) in the shared package are for use by the ringpop, forward, replica and swim packages. These are all separate packages that are inside the ringpop repository.

There's another reason they are in a separate package and not just in ringpop: circular dependencies. Ringpop imports the swim package, which refers to the interfaces. We can't import them back from the ringpop package again. So they have to be in a separate package.

Open to suggestions re: naming.